src/Bundle/DependencyInjection/Driver/Doctrine/DoctrineORMDriver.php

```diff
@@ -44,14 +46,34 @@ protected function addRepository(ContainerBuilder $container, MetadataInterface
             $repositoryClass = $metadata->getClass('repository');
         }
 
+        $aliasId = $metadata->getServiceId('repository');
+        $managerReference = new Reference($metadata->getServiceId('manager'));
         $definition = new Definition($repositoryClass);
-        $definition->setArguments([
-            new Reference($metadata->getServiceId('manager')),
-            $this->getClassMetadataDefinition($metadata),
-        ]);
-        $definition->setPublic(true);
 
-        $container->setDefinition($metadata->getServiceId('repository'), $definition);
+        if ($repositoryClass === EntityRepository::class) {
+            $definition->setFactory([$managerReference, 'getRepository']);
+            $definition->setArguments([$metadata->getClass('model')]);
+            $definition->setPublic(true);
+
+            $container->setDefinition($aliasId, $definition);
+        } else {
+            // we cant use the doctrine factory as this will lead to an endless loop
+            $definition->setArguments([$managerReference, $this->getClassMetadataDefinition($metadata)]);
+            $definition->addTag(ServiceRepositoryCompilerPass::REPOSITORY_SERVICE_TAG);
+
+            $serviceAlias = new Alias($repositoryClass, true);
+            $usesSymfony51Api = method_exists(Alias::class, 'getDeprecation');
+            $deprecationMessage = sprintf('Using the service dot notation "%%alias_id%%" is deprecated. Use the class name "%s"', $repositoryClass);
+
+            if ($usesSymfony51Api) {
+                $serviceAlias->setDeprecated('sylius/resource-bundle', '1.7', $deprecationMessage);
+            } else {
+                $serviceAlias->setDeprecated(true, $deprecationMessage);
+            }
```

Comment:
As we discussed internally, I would prefer to skip the deprecation notice until we will agree on the expected new format of service declaration
```suggestion
```